src/Microsoft.AspNetCore.Http.Connections.Client/Internal/ServerSentEventsTransport.cs

```diff
@@ -116,57 +116,59 @@ public Task StartAsync(Uri url, IDuplexPipe application, TransferFormat transfer
-                var readCancellationRegistration = cancellationToken.Register(
-                    reader => ((PipeReader)reader).CancelPendingRead(), pipelineReader);
+                var options = new PipeOptions(pauseWriterThreshold: 0, resumeWriterThreshold: 0);
+                var reader = PipeReaderFactory.CreateFromStream(options, stream, cancellationToken);
+
                 try
                 {
                     while (true)
                     {
-                        var result = await pipelineReader.ReadAsync();
-                        var input = result.Buffer;
-                        if (result.IsCanceled || (input.IsEmpty && result.IsCompleted))
-                        {
-                            Log.EventStreamEnded(_logger);
-                            break;
-                        }
+                        var result = await reader.ReadAsync(cancellationToken);
+                        var buffer = result.Buffer;
+                        var consumed = buffer.Start;
+                        var examined = buffer.End;
 
-                        var consumed = input.Start;
-                        var examined = input.End;
                         try
                         {
-                            Log.ParsingSSE(_logger, input.Length);
-                            var parseResult = _parser.ParseMessage(input, out consumed, out examined, out var buffer);
-                            FlushResult flushResult = default;
-
-                            switch (parseResult)
+                            if (!buffer.IsEmpty)
                             {
-                                case ServerSentEventsMessageParser.ParseResult.Completed:
-                                    Log.MessageToApp(_logger, buffer.Length);
-
-                                    flushResult = await _application.Output.WriteAsync(buffer);
-
-                                    _parser.Reset();
-                                    break;
-                                case ServerSentEventsMessageParser.ParseResult.Incomplete:
-                                    if (result.IsCompleted)
-                                    {
-                                        throw new FormatException("Incomplete message.");
-                                    }
+                                Log.ParsingSSE(_logger, buffer.Length);
+
+                                var parseResult = _parser.ParseMessage(buffer, out consumed, out examined, out var message);
+                                FlushResult flushResult = default;
+
+                                switch (parseResult)
+                                {
+                                    case ServerSentEventsMessageParser.ParseResult.Completed:
+                                        Log.MessageToApplication(_logger, message.Length);
+
+                                        flushResult = await _application.Output.WriteAsync(message);
```

Comment:
So not using the cancellation token here?

Reply:
No. We use CancelPendingFlush

Reply:
I'd actually prefer to avoid the token in all cases but if the Stream ignores the cancellation token we don't want to hang (I dunno if that happens with all implementations of http client).

Reply:
I'm going to make a slight tweak here to get rid of the CT usage in the pipe itself.

Reply:
Ok did a bit more cleanup here.